CDCgov/prime-reportstream
# Fetch Refactor: Replace NewApi: Cleanup

This PR removes NewApi and directly related files as they have been deprecated and replaced in usage. Includes moving an embedded hook into its own hook file (with test).

Test Steps:
1. Log in
2. Use any organization with senders and receivers (tx-doh, simple-report, etc.)
3. Verify that pages continue to work as expected with data fetches (daily-data, submissions, etc.)

## Changes
- NewApi and directly related files removed
- useAdminSafeOrgName moved and renamed to useAdminSafeOrganizationName (with test)

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #6527
